pkg/services/ngalert/store/image.go

```diff
@@ -0,0 +1,97 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+var (
+	// ErrImageNotFound is returned when the image does not exist.
+	ErrImageNotFound = errors.New("image not found")
+)
+
+type Image struct {
```

Comment:
We should implement `TableName() string` on this struct and return the table name (`alert_image`), this makes sure that we don't need to define the table name below and can just use the struct.

```golang
if _, err := sess.Table("alert_image").Insert(img); err != nil {
    return fmt.Errorf("failed to insert screenshot: %w", err)
}
```
becomes

```golang
if _, err := sess.Insert(img); err != nil {
    return fmt.Errorf("failed to insert screenshot: %w", err)
}
```

Reply:
I like it.

Reply:
That Update path is driving me crazy, though. Every time I tried to just use that update call it would use the wrong key and violate a UNIQUE constraint.